java/server/src/org/openqa/selenium/grid/server/BaseServerOptions.java

```diff
@@ -96,4 +101,24 @@ public URI getExternalUri() {
   public boolean getAllowCORS() {
     return config.getBool("server", "allow-cors").orElse(false);
   }
+
+  public boolean isSecure() {
+    return config.getBool("server", "https").orElse(false);
```

Comment:
Consider `return config.get("server", "https-private-key").isPresent() && config.get("server", "https-certificate").isPresent();`

Reply:
Or throwing a `ConfigException` if one is set and not the other.